tests/src/tests/happy.rs

```diff
@@ -1,13 +1,44 @@
-use consensus_engine::View;
+use consensus_engine::{AggregateQc, Block, Qc, View};
 use fraction::{Fraction, One};
 use futures::stream::{self, StreamExt};
-use std::collections::HashSet;
+use std::fs::OpenOptions;
 use std::time::Duration;
+use std::{collections::HashSet, fs::File};
 use tests::{MixNode, Node, NomosNode, SpawnConfig};
 
 const TARGET_VIEW: View = View::new(20);
 
-async fn happy_test(nodes: Vec<NomosNode>) {
+pub fn persist(name: &'static str, blocks: impl Iterator<Item = &Block>) {
+    // We need to persist the block to disk before dropping it
+    // if there is a aggregated qc when testing happy path
+    if let Qc::Aggregated(qcs) = &block.parent_qc {
+        #[derive(serde::Serialize)]
+        struct Info<'a> {
+            id: String,
+            view: View,
+            qcs: &'a AggregateQc,
+        }
+
+        let mut file = OpenOptions::new()
+            .create(true)
+            .truncate(true)
+            .read(true)
+            .write(true)
+            .open(format!("{name}.json"))
+            .unwrap();
+        let infos = blocks
+            .map(|b| Info {
+                id: format!("{}", header.id),
+                view: header.view,
+                qcs,
+            })
+            .collect::<Vec<_>>();
+        // Use pretty print to make it easier to read, because we need the this for debugging.
+        serde_json::to_writer_pretty(&mut *file, &infos).unwrap();
+    }
+}
```

Comment:
I don't think we should persist them, here are the steps I think need to be done:

* We can add a query for requesting the blocks to the node.
* Before killing the nodes, retrieve blocks.
* Check blocks with this function (or similar).
* Report `NodeId` `BlockId` `View` for every block that is not starndardQc

Reply:
We already have an api to get blocks from the node, I think we can reuse that. In fact, it's the same that we use to check the view, it also has an additional field with blocks info

Reply:
> We already have an api to get blocks from the node, I think we can reuse that. In fact, it's the same that we use to check the view, it also has an additional field with blocks info

Even better!